packages/ra-core/src/controller/input/ReferenceInputBase.spec.tsx

```diff
@@ -0,0 +1,226 @@
+import * as React from 'react';
+import expect from 'expect';
+import { render, screen, waitFor, fireEvent } from '@testing-library/react';
+import { QueryClient } from '@tanstack/react-query';
+import { CoreAdminContext } from '../../core';
+import {
+    ChoicesProps,
+    Form,
+    InputProps,
+    useChoices,
+    useChoicesContext,
+    useInput,
+} from '../../form';
+import { testDataProvider } from '../../dataProvider';
+import { ReferenceInputBase } from './ReferenceInputBase';
+import {
+    SelfReference,
+    QueryOptions,
+    Meta,
+} from './ReferenceInputBase.stories';
+
+describe('<ReferenceInputBase />', () => {
+    const defaultProps = {
+        reference: 'posts',
+        resource: 'comments',
+        source: 'post_id',
+    };
+
+    beforeAll(() => {
+        window.scrollTo = jest.fn();
+    });
+
+    it('should display an error if error is defined', async () => {
+        jest.spyOn(console, 'error')
+            .mockImplementationOnce(() => {})
+            .mockImplementationOnce(() => {});
+
+        render(
+            <CoreAdminContext
+                queryClient={
+                    new QueryClient({
+                        defaultOptions: { queries: { retry: false } },
+                    })
+                }
+                dataProvider={testDataProvider({
+                    getList: () => Promise.reject(new Error('fetch error')),
+                })}
+            >
+                <Form onSubmit={jest.fn()}>
+                    <ReferenceInputBase {...defaultProps}>
+                        <AutocompleteInput />
+                    </ReferenceInputBase>
+                </Form>
+            </CoreAdminContext>
+        );
+        await waitFor(() => {
+            expect(screen.queryByText('fetch error')).not.toBeNull();
+        });
+    });
```

Comment:
I haven't been able to use the `Error` story for this test. Although it behaves as expected in the storybook, it never shows the error in the test. I tried with `fireEvent.change`, `userEvent.type`, with and without blur, with and without additional timeout without success.